### PR TITLE
Feature/us 40 nfr create junits for history flow

### PR DESCRIPTION
**Pull request by JavierMotaAT**

### CODE REVIEW

**Review by EmiliaGuillen-dev — APPROVED** [review]
looks good

File: src/test/java/com/agilethought/atsceapi/service/HistoryServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		when((orikaMapperFacade).mapAsList(anyList(), any())).thenReturn(new ArrayList<>());
+
+		List<HistoryDTO> result = historyService.getUserHistories("15781212");
+
+		assertNotNull(result);
+	}
+
+	@Test
+	public void testCreateHistory() {
+		FixedExpenses fixedExpenses = new FixedExpenses();
```
Comment (ivanflores15) [inline src/test/java/com/agilethought/atsceapi/service/HistoryServiceImplTest.java]: You are repeating code inside HistoryValidatorTest.class. I think is a better way create a dummy factory class in orden to get instances of NewHistoryRequest

File: src/test/java/com/agilethought/atsceapi/validator/history/HistoryValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+	public HistoryValidator historyValidator;
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+
+		populateHistoryRequest();
+	}
+
+	public void populateHistoryRequest() {
```
Comment (ivanflores15) [inline src/test/java/com/agilethought/atsceapi/validator/history/HistoryValidatorTest.java]: Put it inside of dummy factory class

**Review by ivanflores15 — CHANGES_REQUESTED** [review]

**Review by VictorMunguia — APPROVED** [review]

File: src/test/java/com/agilethought/atsceapi/validator/history/HistoryValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+	@Test
+	public void itShouldThrowErrorMessageHistoryInTypeField() {
+
+		Assertions.assertAll(() -> {
+			newHistoryRequest.setType(null);
+			Exception errorMessageException = Assertions.assertThrows(BadRequestException.class, () -> {
+				historyValidator.validate(newHistoryRequest);
+			});
+			Assertions.assertEquals("One or more fields are invalid", errorMessageException.getMessage());
```
Comment (ivanflores15) [inline src/test/java/com/agilethought/atsceapi/validator/history/HistoryValidatorTest.java]: You are using a lot times "One or more fields are invalid", convert to constant or use constant defined in constant class

**Review by ivanflores15 — CHANGES_REQUESTED** [review]

**Review by luisgalindoluna — APPROVED** [review]
Great

**Review by ivanflores15 — APPROVED** [review]

**Pull request merged**